app/stac_api/pgtriggers.py

```diff
@@ -189,6 +196,113 @@ class UpdateItemUpdateIntervalTrigger(pgtrigger.Trigger):
+        of its assets
+    Returns: tuple
+        tuple for all needed triggers
+    '''
+
+    class UpdateCollectionSummariesTrigger(pgtrigger.Trigger):
+        when = pgtrigger.After
+        declare = [
+            ('asset_instance', 'stac_api_collectionasset%ROWTYPE'),
+            ('collection_summaries', 'RECORD'),
+        ]
+        func = '''
+        asset_instance = COALESCE(NEW, OLD);
+
+        -- Compute collection summaries
+        SELECT
+            a.collection_id,
+            array_remove(array_agg(DISTINCT(a.proj_epsg)), null) AS proj_epsg,
+            array_remove(array_agg(DISTINCT(a.geoadmin_variant)), null) AS geoadmin_variant,
+            array_remove(array_agg(DISTINCT(a.geoadmin_lang)), null) AS geoadmin_lang,
+            array_remove(array_agg(DISTINCT(a.eo_gsd)), null) AS eo_gsd
+        INTO collection_summaries
+        FROM (
+            SELECT item.collection_id, asset.proj_epsg, asset.geoadmin_variant, asset.geoadmin_lang, asset.eo_gsd
+            FROM stac_api_item AS item
+                LEFT JOIN stac_api_asset AS asset ON (asset.item_id = item.id)
+            WHERE collection_id = asset_instance.collection_id
+            UNION
+            SELECT collection_id, proj_epsg, geoadmin_variant, geoadmin_lang, eo_gsd
+            FROM stac_api_collectionasset
+            WHERE collection_id = asset_instance.collection_id
+        ) a
+        GROUP BY a.collection_id;
+
+        -- Update related collection (auto variables + summaries)
+        UPDATE stac_api_collection SET
+            updated = now(),
+            etag = gen_random_uuid(),
+            summaries_proj_epsg = collection_summaries.proj_epsg,
+            summaries_geoadmin_variant = collection_summaries.geoadmin_variant,
+            summaries_geoadmin_lang = collection_summaries.geoadmin_lang,
+            summaries_eo_gsd = collection_summaries.eo_gsd
+        WHERE id = asset_instance.collection_id;
+
+        RAISE INFO 'collection.id=% summaries updated, due to collection asset.name=% update.',
+            asset_instance.collection_id, asset_instance.name;
+        RETURN asset_instance;
+        '''
+
+    class UpdateCollectionUpdateIntervalTrigger(pgtrigger.Trigger):
+        when = pgtrigger.After
+        declare = [
+            ('asset_instance', 'stac_api_collectionasset%ROWTYPE'),
+        ]
+        func = '''
+        asset_instance = COALESCE(NEW, OLD);
+        -- Update related collection update_interval variables
+        -- if new value is lower than existing one.
+        UPDATE stac_api_collection SET
+            update_interval = COALESCE(LEAST(NULLIF(asset_instance.update_interval, -1), update_interval), -1)
+        WHERE id = asset_instance.collection_id;
+        RAISE INFO 'collection.id=% update_interval updated, due to collectionasset.name=% updates.',
+            asset_instance.collection_id, asset_instance.name;
+        RETURN asset_instance;
+        '''
```

Comment:
can probably be removed completely, see comment above

Reply:
As we keep the update_interval in the model we should also keep the trigger to avoid issues down the road.